SOC-2188 Update CloseMyAccountMaintenance to use the preferences service

garthwebb opened the pull request:
Also clean up usage of request-closure preference.

https://wikia-inc.atlassian.net/browse/SOC-2188

Commit:
SOC-2188 Newly generated user-pref code

Commit:
SOC-2188 Removed usage of requested-closure

Also added implementation of reverse lookup by user for global preferences.

garthwebb commented on extensions/wikia/EditAccount/SpecialEditAccount_body.php:
```diff
@@ -372,21 +371,20 @@ function setRealName( $realname, $changeReason = '' ) {
 		}
 
-		$id = $user->getId();
-
```
This was unused.

garthwebb commented on lib/Swagger/src/User/Preferences/Models/UserPreferences.php:
```diff
@@ -47,177 +52,208 @@
     }
 }
+
+
```
************\* END GENERATED CODE *************

garthwebb commented on lib/Swagger/composer.json:
```diff
@@ -17,15 +17,15 @@
         }
```
************\* START GENERATED CODE *************

The code from here up to the "END GENERATED CODE" comment below is all generated via swagger definition, feel free to skip down.  If you decide to review it understand that the only way to change this code is to update the swagger definition defined in the `user-preference` service.

Commit:
SOC-2188 Eliminate use of getGlobalAttribute for the close account preference and update tests

Commit:
SOC-2188 Use a constant for the preference name

Commit:
SOC-2188 Create constant for other preference used

garthwebb commented on extensions/wikia/CloseMyAccount/maintenance/CloseMyAccountMaintenance.php:
```diff
@@ -2,6 +2,9 @@
 
```
There are a lot of changes in this file because I replaced the single monolithic `execute` method with several smaller methods.

armonr commented on extensions/wikia/CloseMyAccount/CloseMyAccountHelper.class.php:
```diff
@@ -10,32 +10,27 @@ class CloseMyAccountHelper {
 	const CLOSE_MY_ACCOUNT_WAIT_PERIOD = 30;
 
+	const REQUEST_CLOSURE_PREF = 'requested-closure-date';
+	const DISABLED_BY_USER_PREF = 'disabled-by-user-request';
```
👍

armonr commented on extensions/wikia/CloseMyAccount/maintenance/CloseMyAccountMaintenance.php:
```diff
@@ -15,80 +18,80 @@ public function __construct() {
+		$userObj->setGlobalPreference( CloseMyAccountHelper::DISABLED_BY_USER_PREF, true );
+
+		// Cleanup
+		$userObj->setGlobalPreference( CloseMyAccountHelper::REQUEST_CLOSURE_PREF, null );
```
I think there should also be a call following this to nullify the attribute as well (deleted line 80)

``` php
$userObj->setGlobalAttribute( CloseMyAccountHelper::REQUEST_CLOSURE_PREF, null );
```

garthwebb replied:
See https://wikia-inc.atlassian.net/browse/SOC-2189

armonr commented:
Is `requested-closure` global flag being discarded altogether?

garthwebb commented:
Correct.

On Fri, Jul 8, 2016 at 2:07 PM, Armon Rabiyan notifications@github.com
wrote:

> Is requested-closure global flag being discarded altogether?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> https://github.com/Wikia/app/pull/10805#issuecomment-231473045, or mute
> the thread
> https://github.com/notifications/unsubscribe/AAEbfoNeCkQRO-3pvw4iE53pUrSWUMz1ks5qTrwegaJpZM4JHlVi
> .

## 

Garth Webb

_Engineering Manager_

e  garth@fandom.com

m 415-652-7688

w  fandom.com

Wikia, Inc.

360 3rd Street, Suite 750

San Francisco, CA 94107

[image: EmailSignature-05.png]

https://twitter.com/getfandom https://www.facebook.com/getfandom/
https://www.instagram.com/getfandom/
https://www.linkedin.com/company/wikia

A Wikia, Inc. Property.

armonr commented:
Besides failing tests, LGTM

Commit:
SOC-2188 Make sure test don't complain about an abstract method that hasn't been defined in the mock.

Commit:
SOC-2188 Get rid of test data

This test data only made sense when we had both requested_closure and requested_closure_date in use at the same time.  Now that there is only one source of truth ('requested_closure_date') there's no need to test this extra combination.

armonr commented:
:shipit:

jsutterfield commented:
LGTM!
